src/coreclr/System.Private.CoreLib/src/System/Enum.CoreCLR.cs

```diff
@@ -13,8 +13,35 @@ public abstract partial class Enum
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Enum_GetValuesAndNames")]
         private static partial void GetEnumValuesAndNames(QCallTypeHandle enumType, ObjectHandleOnStack values, ObjectHandleOnStack names, Interop.BOOL getNames);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern object InternalBoxEnum(RuntimeType enumType, long value);
+        private static unsafe object InternalBoxEnum(RuntimeType enumType, long value)
+        {
+            MethodTable* pMethodTable = (MethodTable*)enumType.m_handle;
+            object obj = GetEnumInfo(enumType, false).CreateUninitializedInstance(enumType);
```

Comment:
This is also going to fetch the enum values - it is more than what we need here.

Reply:
I've reworked the logic in a6839324e49862f136c966a5f8b9069bff092a78 to only ever initialize the reflection-related info when needed, and otherwise only initialize the allocator info. Both are now initialized separately, but existing `EnumInfo` instances are otherwise reused. Hope this is better! 🙂